test/type/implicit_some/opaque_return.swift

```diff
@@ -103,3 +103,11 @@ func eat(_ myfruit: inout Basket) -> Basket {
   return myfruit
 }
 
+protocol P {
+  associatedtype A: P
+  var value: A { get }
+}
+
+struct S: P {
+  var value: P { self }
+}
```

Comment:
@angela-laar or @hborla, it would be nice to add some tests for parameterized protocol types here too (in a future PR)